src/collectors/windows.plugin/perflib-mssql.c

```diff
@@ -730,17 +791,667 @@ static void do_mssql_errors(PERF_DATA_BLOCK *pDataBlock, struct mssql_instance *
+        perflibGetObjectCounter(pDataBlock, pObjectType, &mdi->MSSQLDatabaseActiveTransactions);
+        perflibGetObjectCounter(pDataBlock, pObjectType, &mdi->MSSQLDatabaseBackupRestoreOperations);
+        perflibGetObjectCounter(pDataBlock, pObjectType, &mdi->MSSQLDatabaseLogFlushed);
+        perflibGetObjectCounter(pDataBlock, pObjectType, &mdi->MSSQLDatabaseLogFlushes);
+        perflibGetObjectCounter(pDataBlock, pObjectType, &mdi->MSSQLDatabaseTransactions);
+        perflibGetObjectCounter(pDataBlock, pObjectType, &mdi->MSSQLDatabaseWriteTransactions);
+    }
+
+    dictionary_sorted_walkthrough_read(p->databases, dict_mssql_databases_charts_cb, &update_every);
+    rrdset_done(p->st_db_data_file_size);
+    rrdset_done(p->st_db_active_transactions);
+    rrdset_done(p->st_db_backup_restore_operations);
+    rrdset_done(p->st_db_log_flushed);
+    rrdset_done(p->st_db_log_flushes);
+    rrdset_done(p->st_db_transactions);
+    rrdset_done(p->st_db_write_transactions);
+}
+
+static void do_mssql_memory_mgr(PERF_DATA_BLOCK *pDataBlock, struct mssql_instance *p, int update_every)
+{
+    char id[RRD_ID_LENGTH_MAX + 1];
+    PERF_OBJECT_TYPE *pObjectType = perflibFindObjectTypeByName(pDataBlock, p->objectName[NETDATA_MSSQL_MEMORY]);
+    if (!pObjectType)
+        return;
+
+    if (perflibGetObjectCounter(pDataBlock, pObjectType, &p->MSSQLConnectionMemoryBytes)) {
+        snprintfz(id, RRD_ID_LENGTH_MAX, "instance_%s_memmgr_connection_memory_bytes", p->instanceID);
+        if (!p->st_conn_memory) {
+            p->st_conn_memory = rrdset_create_localhost("mssql"
+                                                        , id, NULL
+                                                        , "memory"
+                                                        , "mssql.instance_memmgr_connection_memory_bytes"
+                                                        , "Amount of dynamic memory to maintain connections"
+                                                        , "bytes"
+                                                        , PLUGIN_WINDOWS_NAME
+                                                        , "PerflibMSSQL"
+                                                        , PRIO_MSSQL_MEMMGR_CONNECTION_MEMORY_BYTES
+                                                        , update_every
+                                                        , RRDSET_TYPE_LINE
+                                                        );
+
+            snprintfz(id, RRD_ID_LENGTH_MAX, "mssql_instance_%s_memmgr_connection_memory_bytes", p->instanceID);
+            p->rd_conn_memory  = rrddim_add(p->st_conn_memory,
+                                           id,
+                                           "memory",
+                                           1,
+                                           1,
+                                           RRD_ALGORITHM_INCREMENTAL);
```

Comment:
Is this incremental?

Reply:
Thank you @stelfrag. We were using `incremental` in our `go.d.plugin`, but they are gauge.